### DISCUSSION TITLE
queryBuilder: how do I select all non-virtual properties from table?

### DISCUSSION THREAD

**Discussion by squivix** [mikro-orm/mikro-orm#5305] [Q&A] [answered]
I need to select all columns from table. No virtual properties, computed properties, or formulas. How do I do this with the query builder? I'd write a raw query but I don't know how to get the postgres schema name

**Comment by B4nan** [comment]
What's the use case? Maybe you can just use `FindOptions.exclude` to skip them? Or explicit serialization? 

> I'd write a raw query but I don't know how to get the postgres schema name

Not sure what you mean by this?

**Comment by squivix** [comment]
The use case is dumping the entire database into JSON line files with properties being raw column names.

> Not sure what you mean by this?

I mean running a raw query like this:
```ts
const batch = await em.execute(`select * from ${em.getMetadata(entityClass).tableName} limit ${batchSize} offset ${insertedCount}`);
```

but then how do I qualify the table name with the postgres schema name? ie. `public.user` is different from the db's own `user`

I gues I could hardcode it, but is there a way to get the current schema from the `em`? I found  `em.schema` but it's undefined.

**Reply by B4nan** [reply, marked as answer]
> The use case is dumping the entire database into JSON line files with properties being raw column names.

That's not really something you should use an ORM for if you ask me (let alone the idea of dumping everything into a single JSON file seems quite courageous :]).

> but then how do I qualify the table name with the postgres schema name? ie. public.user is different from the db's own user

Default schema is `public` or the one you set in the ORM config - which you can easily get from the ORM config on runtime as well.

> I found em.schema but it's undefined.

That's an override on EM level, you can use `em.config.get('schema')` for the global default, or there is also `em.getDriver().getSchemaName()` which does part of this for you (checking for wildcard schema and schema override on metatada level, falling back to the ORM config value).

**Reply by squivix** [reply]
> That's not really something you should use an ORM for if you ask me (let alone the idea of dumping everything into a single JSON file seems quite courageous :]).

You're probably right, but I'm already familiar with it so why not? Also it's one [JSON line](https://jsonlines.org/) file per table, not one JSON file for the full dataset. Reading is streaming and parsing one line at a time, collecting into batches and inserting back to db. I prefer this over SQL dumps because it's more readable.

Thanks for the help

**Reply by B4nan** [reply]
> why not

Performance mostly, if it works fast enough for you, it's ok.
